### DISCUSSION TITLE
Why my custom layout is embed when I have a app/router.options.ts file ?

### DISCUSSION THREAD

**Discussion by jfsenechal** [nuxt/nuxt#19700] [Questions] [answered]
Hi,

When I have a router.options.ts file in app directory, my page "about" with home layout 
is embed in default.layout

When I remove app/router.options.ts file , my page "about" is only with home layout

```vue
<!--pages/about.vue-->
<script setup>
definePageMeta({
    layout: false //remove default layout, this is not working when file app/router.options.ts is present
})
</script>
```

https://github.com/jfsenechal/nuxt-layouts

**Comment by danielroe** [comment, marked as answer]
You are overriding the default routes, so that also means you are overriding the default route _metadata_. If you want to do this, you will need to set `meta: { layout: false }` manually in your `router.options.ts`. Or you can read the default metadata from the array of routes that is passed to the function.

**Comment by jfsenechal** [comment]
Thanks @danielroe

I created this file because my routing rules are too complex to put in this file name in the "pages" directory

But I can no longer use definePageMeta in my vue files

```vue
definePageMeta({
  layout: false,
  title: 'xxx',
})
```

How can I continue to use them ?

**Comment by JuliWolf** [comment]
Hello
have the same problem with `definePageMeta` after setting custom `router.oprions.ts`

Now it is imposible to set page middleware, layout, validation and etc
Middleware also does not work if set them directrly in routes.meta

```
// routes
[
 ...
 { name: 'someName', component: SomeComponent, meta: { middleware: ['testMiddleware' ] } }
 ...
]
```

**Comment by jfsenechal** [comment]
Hi

Add default layout: false in your route
And add   <NuxtLayout name="homelayout"> in your temlate


```ts
// routes
[
 ...
 { name: 'someName', component: SomeComponent, meta: { layout:false,middleware: ['testMiddleware' ] } }
 ...
]
```

```vue

<template>
  <NuxtLayout name="homelayout">
....
  </NuxtLayout>
</template>
```

**Reply by JuliWolf** [reply]
When I set layout in meta in routes it works fine even without adding NuxtLayout in template
```
// routes
[
 ...
 { name: 'someName', component: SomeComponent, meta: { layout: 'myLayout' } }
 ...
]
```
the main problem is with page middlewares, validators and other `usePageMeta` options
